tests/integration/helpers/loginByGoogle.js

```diff
@@ -1,45 +1,54 @@
+import { selectLocaleFromDialog } from "./selectLocale";
+
+const axios = require("axios");
 
 const setWindowAccessToken = ClientFunction((data) =>
   window.__store__.dispatch("auth/setAccessToken", data)
 );
 
 export const googleAuthUser = Role(
   `${process.env.BROWSERSTACK_BASE_URL}/login`,
-  async (t) => {
-    await fetch("https://www.googleapis.com/oauth2/v4/token", {
-      method: "POST",
-      body: JSON.stringify({
-        grant_type: "refresh_token",
-        client_id: process.env.GOOGLE_OAUTH2_CLIENT_ID,
-        client_secret: process.env.GOOGLE_OAUTH2_CLIENT_SECRET,
-        refresh_token: process.env.GOOGLE_OAUTH2_REFRESH_TOKEN,
-      }),
-      headers: { "Content-Type": "application/json" },
-    })
-      .then((res) => res.json())
-      .then(async (data) => {
-        const { access_token: socialAuthToken } = data;
+  async (testcafe) => {
+    let headers = { "Content-Type": "application/json" };
+    await axios
+      .post(
+        "https://www.googleapis.com/oauth2/v4/token",
+        JSON.stringify({
+          grant_type: "refresh_token",
+          client_id: process.env.GOOGLE_OAUTH2_CLIENT_ID,
+          client_secret: process.env.GOOGLE_OAUTH2_CLIENT_SECRET,
+          refresh_token: process.env.GOOGLE_OAUTH2_REFRESH_TOKEN,
+        }),
+        { headers }
+      )
+      .then(async (response) => {
+        const socialAuthToken = response.data.access_token;
 
-        await fetch(process.env.DJANGO_CONVERT_SOCIAL_AUTH_TOKEN_URL, {
-          method: "POST",
-          body: JSON.stringify({
-            grant_type: "convert_token",
-            client_id: process.env.DJANGO_DEFAULT_OAUTH2_CLIENT_ID,
-            client_secret: process.env.DJANGO_DEFAULT_OAUTH2_CLIENT_SECRET,
-            backend: "google-oauth2",
-            token: socialAuthToken,
-          }),
-          headers: { "Content-Type": "application/json" },
-        })
-          .then((res) => res.json())
-          .then(
-            async (data) => {
-              await setWindowAccessToken(data);
-            },
-            () => {}
-          );
+        await axios
+          .post(
+            process.env.BACKEND_CONVERT_SOCIAL_AUTH_TOKEN_URL,
+            JSON.stringify({
+              grant_type: "convert_token",
+              client_id: process.env.BACKEND_DEFAULT_OAUTH2_CLIENT_ID,
+              client_secret: process.env.BACKEND_DEFAULT_OAUTH2_CLIENT_SECRET,
+              backend: "google-oauth2",
+              token: socialAuthToken,
+            }),
+            { headers }
+          )
+          .then(async (response) => {
+            await setWindowAccessToken(response.data);
+          });
       });
   }
 );
+
+export const loginGoogleUserAndSetLocale = async (locale = "en") => {
+  await testcafe
+    .useRole(googleAuthUser)
+    .navigateTo(`${process.env.BROWSERSTACK_BASE_URL}/home`);
+
+  // have to set the locale again as the login page is loaded in a different language
```

Comment:
I think since this helper function is never seeing the beforeEach function in each test file where the locale is being set, this `again` here is causing some confusion. let's just remove the comment?